agent/tool-scripts/pprof

```diff
@@ -105,79 +104,40 @@ tool_cmd_file="$tool_output_dir/$tool.cmd"
-                    fi
-                else
-                    if grep -q "^OPENSHIFT_PROFILE=web" $ose_openshift_master; then
-                        systemctl restart atomic-openshift-master
-                    else
-                        echo "OPENSHIFT_PROFILE=web" >> $ose_openshift_master
-                        systemctl restart atomic-openshift-master.service
-                    fi
-                fi
-            fi 
-            ;;
-        node)
-            if [ -e $origin-node ]; then 
-                if grep -q "^OPENSHIFT_PROFILE=web" $origin_node; then
-                    systemctl restart openshift-node.service
-                else
-                    echo "OPENSHIFT_PROFILE=web" >> $origin_node
-                    systemctl restart openshift-node.service
-                fi 
-            else
-                if grep -q "^OPENSHIFT_PROFILE=web" $ose_openshift_node; then
-                    systemctl restart atomic-openshift-node.service
-                else
-                    echo "OPENSHIFT_PROFILE=web" >> $ose_openshift_node
-                    systemctl restart atomic-openshift-node.service
-                fi
-            fi 
-    esac
-}
 
 collect_data() {
     while true; do
         if [ $(systemctl list-unit-files | grep atomic-openshift-master.service | awk {'print $2'}) == "masked" ]; then
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_controller.txt http://localhost:6060/debug/pprof/profile 
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_controller_heap.txt http://localhost:6060/debug/pprof/heap 
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_api.txt http://localhost:6061/debug/pprof/profile
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_api_heap.txt http://localhost:6061/debug/pprof/heap 
-        else
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S).txt http://localhost:6060/debug/pprof/profile 
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_heap.txt http://localhost:6060/debug/pprof/heap 
+            for host in "${hosts[@]}"; do
+                echo $host
+                oc get --raw /debug/pprof/profile ${host} > profile && $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/${host}/$(date +%Y-%m-%d:%H:%M:%S).txt profile
+                oc get --raw /debug/pprof/heap ${host} > heap && $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/${host}/$(date +%Y-%m-%d:%H:%M:%S)_heap.txt heap
+            done
         fi
-    done 
+    done
 } 
+
+## check if the inventory variable is set correctly
+if [[ -z "$inventory" ]]; then
+	usage
```

Comment:
usage function is not defined

Reply:
Nice spot, I just lifted these conditionals for inventory from your `prometheus-metrics` script.